packages/contractkit/src/identity/offchain-data-wrapper.test.ts

```diff
@@ -145,10 +163,102 @@ testWithGanache('Offchain Data', (web3) => {
       const nameAccessor = new NameAccessor(wrapper)
       await nameAccessor.write({ name: testname })
 
-      const resp = await nameAccessor.readAsResult(writer)
+      const resp = await nameAccessor.read(writerAddress)
       if (resp.ok) {
         expect(resp.result.name).toEqual(testname)
       }
     })
   })
+
+  describe('encryption', () => {
+    describe('when no keys are loaded in the wallet', () => {
+      it('cannot write encrypted data', async () => {
+        const testname = 'test'
+        const payload = { name: testname }
+
+        const nameAccessor = new NameAccessor(wrapper)
+
+        try {
+          await nameAccessor.writeEncrypted(payload, writerAddress, readerAddress)
+          throw new Error("Shouldn't get here")
+        } catch (e) {
+          expect(e.message).toEqual(`Could not find address ${writerAddress.toLowerCase()}`)
+        }
+      })
+
+      it('the reader cannot decrypt the data', async () => {
+        const testname = 'test'
+        const payload = { name: testname }
+
+        const nameAccessor = new NameAccessor(wrapper)
+        kit.addAccount(writerPrivate)
+        await nameAccessor.writeEncrypted(payload, writerAddress, readerAddress)
+
+        try {
+          await nameAccessor.readEncrypted(writerAddress, readerAddress)
+          throw new Error('Should not get here')
+        } catch (e) {
+          expect(e.message).toEqual(`Could not find address ${readerAddress.toLowerCase()}`)
+        }
+
+        kit.removeAccount(writerAddress)
+      })
+    })
+
+    describe('using keys in the wallet', () => {
```

Comment:
It seems like there should be two separate cases here?
1. The writer has the key accessible, but the reader doesn't
2. Both have their key accessible?